core/store/src/main/java/org/locationtech/geowave/core/store/base/BaseDataStore.java

```diff
@@ -1252,37 +1158,137 @@ public void copyTo(
+					else {
+						// mark the index for removal
+						markedAdapters.add(dataAdapter.getAdapterId());
+					}
+				}
+			}
+		}
+		it.close();
+
+		// take out the index from the mapping, statistics, and data
+		for (int i = 0; i < markedAdapters.size(); i++) {
+			short adapterId = markedAdapters.get(i);
+
+			// delete the data and the stats for this adapter/index combination
+			final Query q = QueryBuilder.newBuilder().addTypeName(
+					internalAdapterStore.getTypeName(adapterId)).indexName(
+					indexName).build();
+			internalQuery(
+					q,
+					true);
+			indexMappingStore.remove(
+					adapterId,
+					indexName);
+		}
+		// remove the actual index
+		indexStore.remove(indexName);
 	}
 
 	@Override
-	public void removeIndex(
-			final String typeName,
-			final String indexName ) {
-		// TODO Auto-generated method stub
-
+	public void removeIndexMapping(
+			String typeName,
+			String indexName )
+			throws AdapterException, IllegalArgumentException {
+		short adapterId = internalAdapterStore.getAdapterId(typeName);
+
+		final AdapterToIndexMapping adapterIndexMap = indexMappingStore.getIndicesForAdapter(adapterId);
+		if(adapterIndexMap == null) {
+			throw new IllegalArgumentException("No adapter with typeName " + typeName + "could be found.");
+		}
+		
+		final String[] indexNames = adapterIndexMap.getIndexNames();
+		if (indexNames.length == 1) {
+			throw new AdapterException(
+					"Index mapping removal failed. Adapters require at least one index.");
+		}
+		else {
+			// delete the data and the stats for this adapter/index combination
+			final Query q = QueryBuilder.newBuilder().addTypeName(
+					typeName).indexName(
+					indexName).build();
+			internalQuery(
+					q,
+					true);
+
+			// clean up the index/adapter mapping
+			indexMappingStore.remove(
+					adapterId,
+					indexName);
+		}
 	}
 
 	@Override
 	public void removeType(
-			final String typeName ) {
-		// TODO Auto-generated method stub
+			String typeName ) {
+		// Removing a type requires removing the data associated with the type,
+		// the index mapping
+		// for the type, we also need to remove stats for the type.
+		final Short adapterId = internalAdapterStore.getAdapterId(typeName);
+
+		if (adapterId != -1) {
+			final AdapterToIndexMapping mapping = indexMappingStore.getIndicesForAdapter(adapterId);
+			final String[] indexNames = mapping.getIndexNames();
+
+			// remove all the data for each index paired to this adapter
+			for (int i = 0; i < indexNames.length; i++) {
+				baseOperations.deleteAll(
+						indexNames[i],
+						adapterId);
+			}
 
+			statisticsStore.removeAllStatistics(adapterId);
+			indexMappingStore.remove(adapterId);
+			internalAdapterStore.remove(adapterId);
+			adapterStore.removeAdapter(adapterId);
+		}
 	}
 
 	@Override
 	public void deleteAll() {
-		// TODO Auto-generated method stub
-
+		// delete all the data for every adapter and every index
+		try {
+			// passing null deletes everything!
+			delete(null);
```

Comment:
this method was intended for `deleteEverything()` I thought (a convenience method for not just deleting the data but also the metadata.